Joystream-Role-Bot/src/controls/control.ts

```diff
@@ -150,13 +155,23 @@ export const setMemberRole = async (client: Client): Promise<void> => {
 
       const roleUpdatePromise =
         memberRole.status.__typename === "WorkerStatusActive"
-          ? discordMember.roles.add(role)
-          : discordMember.roles.remove(role);
+          ? await discordMember.roles.add(role)
+          : await discordMember.roles.remove(role);
+
       await roleUpdatePromise;
+
+      const daoRole = await guild.roles.fetch(RoleAddress.DAO);
+
+      if (!daoRole) {
+        console.log(`<@&${daoRole}> Role not found`);
+        return;
+      }
+
+      (memberRole.isLead || memberRole.status.__typename === "WorkerStatusActive" || qnMember.isCouncilMember) ? await discordMember.roles.add(daoRole) : await discordMember.roles.remove(daoRole);
```

Comment:
This isn't very readable, let's use `if-else`

Reply:
this don't need on the second #30